optimum/intel/pipelines/__init__.py

```diff
@@ -0,0 +1,411 @@
+#  Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import warnings
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union
+
+from huggingface_hub import model_info
+from transformers.configuration_utils import PretrainedConfig
+from transformers.models.auto.configuration_auto import AutoConfig
+from transformers.models.auto.tokenization_auto import TOKENIZER_MAPPING, AutoTokenizer
+from transformers.pipelines.base import (
+    Pipeline,
+    PipelineRegistry,
+    get_default_model_and_revision,
+)
+from transformers.pipelines.text_generation import TextGenerationPipeline
+from transformers.tokenization_utils import PreTrainedTokenizer
+from transformers.utils import (
+    CONFIG_NAME,
+    HUGGINGFACE_CO_RESOLVE_ENDPOINT,
+    cached_file,
+    extract_commit_hash,
+    is_ipex_available,
+    is_offline_mode,
+    is_torch_available,
+    logging,
+)
+
+from ..generation.modeling import TSModelForCausalLM
+
+
+if is_ipex_available():
+    import intel_extension_for_pytorch
+
+
+if is_torch_available():
+    import torch
+    from transformers.models.auto.modeling_auto import AutoModelForCausalLM
+
+
+if TYPE_CHECKING:
+    from transformers.modeling_utils import PreTrainedModel
+    from transformers.tokenization_utils_fast import PreTrainedTokenizerFast
+
+
+logger = logging.get_logger(__name__)
+
+
+# Register all the supported tasks here
+TASK_ALIASES = {
+    "sentiment-analysis": "text-classification",
+}
+SUPPORTED_TASKS = {
+    "text-generation": {
+        "impl": TextGenerationPipeline,
+        "pt": (AutoModelForCausalLM,) if is_torch_available() else (),
+        "default": {"model": {"pt": ("gpt2", "6c0e608")}},
+        "type": "text",
+    },
+}
+
+
+PIPELINE_REGISTRY = PipelineRegistry(supported_tasks=SUPPORTED_TASKS, task_aliases=TASK_ALIASES)
+
```

Comment:
The transformers pipelines are already compatible with our models, so I'm not sure it makes sense to create new pipelines just to hide the export phase, also we should think about other format (like openvino), it could be very unclear for users what the `optimum.intel.pipelines` are doing

Reply:
You were right. I propose that use `optimum.intel.pipeline` to replace `optimum.intel.ipex.inference`, but we can consider `Neural compressor` and `openvino` in the pipeline. What do you think?

Reply:
I see your point, then I think we can just have "ipex", "openvino" as [accelerator](https://github.com/huggingface/optimum/blob/v1.18.0/optimum/pipelines/pipelines_base.py#L296) like done in the `optimum.pipelines`

Reply:
I agree that we should deprecate `optimum.intel.ipex.inference`

Reply:
> It is a good idea. I can implement the ipex accelerator and leave the OV and INC to their owner : )

Works for me!

> You were right. I propose that use `optimum.intel.pipeline` to replace `optimum.intel.ipex.inference`, because wrap is not very clear for users and ipex didn't have a model optimization way like openvino: `model = OVModelForSeq2SeqLM.from_pretrained(model_id)`.

Could you deprecate it then https://github.com/huggingface/optimum-intel/blob/v1.16.0/optimum/intel/ipex/inference.py#L79 : add a warning to state support will be removed in a future release of optimum-intel ? what do you think about v1.18.0